### PR TITLE
kvm: follow up to #1525 issue

### PR DESCRIPTION
**Pull request by borqosky**
Used fixed data directory for kvm keys Move the ssh key generation
from stage0 to stage1 and remove port from sshd config file

### CODE REVIEW

**Comment by jellonek** [comment]
Code is ok, but commit message needs improvement. We will work on this on Monday morning...

**Comment by jellonek** [comment]
LGTM. My proposal for commit message:

> kvm: follow up to #1525 issue
> 
> ssh keys setup moved to stage1
> data directory (as requested by @krnowak) separated from stage0 rkt datadir
> sshd port removed from configuration (was unused, inetd mode)

File: stage1/enter_kvm/enter_kvm.go [hunk 1]
```diff
@@ -38,6 +40,92 @@ var (
...
+func fileAccessible(path string) bool {
+	if info, err := os.Stat(path); err == nil {
+		return info.Mode().IsRegular()
+	}
+	return false
+}
+
+func sshPrivateKeyPath() string {
+	return filepath.Join(kvmSettingsDir, kvmPrivateKeyFilename)
+}
```
Comment (ppalucki) [inline stage1/enter_kvm/enter_kvm.go:57]: you can: `new const = const + const`

File: stage1/enter_kvm/enter_kvm.go [hunk 2]
```diff
@@ -38,6 +40,92 @@ var (
...
+
+func sshPublicKeyPath() string {
+	return sshPrivateKeyPath() + ".pub"
```
Comment (ppalucki) [inline stage1/enter_kvm/enter_kvm.go:60]: the same as for previous

File: stage1/enter_kvm/enter_kvm.go [hunk 3]
```diff
@@ -38,6 +40,92 @@ var (
...
+
+		if err := generateKeyPair(private); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func ensureAuthorizedKeysExist(keyDirPath string) error {
+	fout, err := os.OpenFile(
```
Comment (ppalucki) [inline stage1/enter_kvm/enter_kvm.go:95]: why not os.Create ?

File: stage1/enter_kvm/enter_kvm.go [hunk 4]
```diff
@@ -38,6 +40,92 @@ var (
...
+func sshPrivateKeyPath() string {
+	return filepath.Join(kvmSettingsDir, kvmPrivateKeyFilename)
+}
+
+func sshPublicKeyPath() string {
+	return sshPrivateKeyPath() + ".pub"
+}
+
+// generateKeyPair calls ssh-keygen with private key location for key generation purpose
+func generateKeyPair(private string) error {
```
Comment (ppalucki) [inline stage1/enter_kvm/enter_kvm.go:64]: Wlodek - (your idea) is it worth to replace that `crypto/rsa.Generate` ?

File: stage1/enter_kvm/enter_kvm.go (hunk 1 above)
Comment (borqosky) [inline stage1/enter_kvm/enter_kvm.go:57]: yes, but you need to worry to add "/", or not after each const. Seems, that there also cannot use filepath.Join (seems reasonable that every function) in const area

File: stage1/enter_kvm/enter_kvm.go (hunk 3 above)
Comment (borqosky) [inline stage1/enter_kvm/enter_kvm.go:95]: Guess that ssh files need to have some different mode to use it...

File: stage1/enter_kvm/enter_kvm.go (hunk 4 above)
Comment (borqosky) [inline stage1/enter_kvm/enter_kvm.go:64]: IMHO can be next follow up, after quick research

Comment (jellonek) [inline stage1/enter_kvm/enter_kvm.go:64]: Good idea to do it later (and also use ssh lib to make connection, what would remove ssh as client dependency).

**Pull request merged**